backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/75-de.metas.fresh.reports/5594360_sys_AddGLN_ToAddresInFunctionsForDocs.sql

```diff
@@ -0,0 +1,202 @@
+    city            text,
+    country         text,
+    gln             text,
+    AddressBlock    text
+);
+
+
+
+CREATE OR REPLACE FUNCTION de_metas_endcustomer_fresh_reports.docs_generics_bpartner_report(org_id numeric,
+                                                                                            doctype text,
+                                                                                            bp_loc_id numeric,
+                                                                                            record_id numeric)
+    RETURNS SETOF de_metas_endcustomer_fresh_reports.docs_generics_bpartner_report AS
+$BODY$
+SELECT x.org_name,
+       x.org_addressline,
+       x.address1,
+       x.postal,
+       x.city,
+       x.country,
+       bpl.gln,
+
+       CASE
+           WHEN $2 = 'o'
+               THEN o.BPartnerAddress
+           WHEN $2 = 'o_delivery'
+               THEN o.DeliveryToAddress
+           WHEN $2 = 'io'
+               THEN io.BPartnerAddress
+           WHEN $2 = 'freshio'
+               THEN freshio.BPartnerAddress
+           WHEN $2 = 'i'
+               THEN i.BPartnerAddress
+           WHEN $2 = 'l'
+               THEN tl.BPartnerAddress
+           WHEN $2 = 'lt'
+               THEN COALESCE(bpg.name || ' ', '') || COALESCE(bp.name || ' ', '') ||
+                    COALESCE(bp.name2 || E'\n', E'\n') ||
+                    COALESCE(letter.BPartnerAddress, '')
+           WHEN $2 = 'd'
+               THEN d.BPartnerAddress
+           WHEN $2 = 'rfqr'
+               THEN COALESCE(bprfqr.name || E'\n', '') || COALESCE(bplrfqr.address, '')
+           WHEN $2 = 'ft'
+               THEN COALESCE(bpft.name || E'\n', '') || COALESCE(bplft.address, '')
+           WHEN $2 = 'mkt'
+               THEN COALESCE(mktbp.name || E'\n', '') || COALESCE(mktbpl.address, '')
+           WHEN $2 = 'ci'
+               THEN ci.BPartnerAddress
+           WHEN $3 IS NOT NULL
+               THEN
+               COALESCE(bp.name || E'\n', '') || COALESCE(bpl.address, '')
+           ELSE 'Incompatible Parameter!'
+           END || E'\n' AS addressblock
+FROM (
+         SELECT COALESCE(org_bp.name, '')  as org_name,
+                trim(
+                                    COALESCE(org_bp.name || ', ', '') ||
+                                    COALESCE(loc.address1 || ' ', '') ||
+                                    COALESCE(loc.postal || ' ', '') ||
+                                    COALESCE(loc.city, '')
+                    )                      as org_addressline,
+                COALESCE(loc.address1, '') as address1,
+                COALESCE(loc.postal, '')   as postal,
+                COALESCE(loc.city, '')     as city,
+                c.Name                     as country
+         FROM ad_orginfo oi
+                  JOIN c_bpartner_location org_bpl
+                       ON org_bpl.c_bpartner_location_ID = oi.orgbp_location_id AND org_bpl.isActive = 'Y'
```

Comment:
I think there shouldn't be an IsActive=Y condition. There is just one org_bpl, and it's needed as per the parameter.
=> on a quick glimpse, this applied to almost all `.isActive = 'Y'`conditions that we have here

Btw, can you also please prepend `p_`to the parameters and then replace the `$n` with param-names?